sundog/view/documents.py

```diff
@@ -0,0 +1,329 @@
+from datatableview import Datatable
+from datatableview.columns import CompoundColumn, DisplayColumn, TextColumn
+from datatableview.helpers import make_processor, through_filter
+from datatableview.views import XEditableDatatableView
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
+from django.db.models import (
+    DateTimeField,
+    ForeignKey,
+    Model,
+    SET_NULL,
+)
+from django.forms.models import ModelForm
+from django.forms.widgets import Select, SelectMultiple
+from django.template.defaultfilters import date
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.views.generic.edit import UpdateView
+from fm.views import AjaxCreateView, AjaxDeleteView, AjaxUpdateView
+from multiselectfield import MultiSelectField
+from settings import SHORT_DATETIME_FORMAT
+from sundog.models import NONE_CHOICE_LABEL
+from sundog.routing import decorate_view, route
+from sundog.utils import LongCharField
+from tinymce.models import HTMLField
+
+
+class Document(Model):
```

Comment:
As discussed, I believe splitting `modules.py` into smaller chunks is preferable as it helps avoid increased merge conflicts due to gigantic files (as witnessed by my last two painful days spent merging `views.py`).

I see several ways forward:

1.  We could keep section-specific models together with the views for the section in a file within `sundog/view/`, as done in this PR.  This works fine, but it’s indeed unusual practice (although not for technical limitations) for models and views to inhabit the same file in Django, and for sections with many models, it could lead to large files.

2.  We could move section-specific models to separate files just for the models related to the section; in this case, something like `sundog/documents/models.py` with just this one model, for now.  Then the views would go in `sundog/documents/views.py`, and likewise for forms and the like.  This might need another top-level directory, like `sundog/sections/` or such, to contain it all, as the discovery mechanism might need this to work properly; the `sundog/view/` directory takes care of this currently, although perhaps it should have a better name, as it also contains models, forms, and the like — not just views.

3.  We could organize the system into separate Django applications.  This is a rather large hammer, though, and would require considerable refactoring.  I don’t know enough about Django applications to figure out the benefits of going this route over the previous.

4.  We could have the dual of route 2, which would have separate toplevel packages `sundog/model/` and `sundog/view/` wherein would live small files with the models and the views for specific sections of the application, such as `sundog/model/documents.py` and `sundog/view/documents.py`.  This is similar to route 1, but instead of all _related_ code being close, you would have all _similar_ code being close.  This is closer to the traditional Django way of things.

5.  We could jump on the microservice bandwagon and have a beautifully overengineered application that will never be finished ;-)

I *prefer* routes 1 or 2, as I believe it’s easier to manage complexity when [everything is related to everything else, but near things are more related than distant things](https://en.wikipedia.org/wiki/Tobler%27s_first_law_of_geography).  But route 4 is, of course, very much equivalent to route 2, except for where things end up in the filesystem.

Thoughts?